CMakeLists.txt

```diff
@@ -263,145 +264,159 @@ endif ()
+  endif ()
+  if (HAVE_SSE2)
+    target_compile_options (${fftw3_lib} PRIVATE ${SSE2_FLAG})
+  endif ()
+  if (HAVE_AVX)
+    target_compile_options (${fftw3_lib} PRIVATE ${AVX_FLAG})
+  endif ()
+  if (HAVE_AVX2)
+    target_compile_options (${fftw3_lib} PRIVATE ${AVX2_FLAG})
+  endif ()
+  if (HAVE_FMA)
+    target_compile_options (${fftw3_lib} PRIVATE ${FMA_FLAG})
+  endif ()
+  if (HAVE_LIBM)
+    target_link_libraries (${fftw3_lib} m)
+  endif ()
 
-add_library (${fftw3_lib} ${SOURCEFILES})
-if (MSVC)
-  target_compile_definitions (${fftw3_lib} PRIVATE /bigobj)
-endif ()
-if (HAVE_SSE)
-  target_compile_options (${fftw3_lib} PRIVATE ${SSE_FLAG})
-endif ()
-if (HAVE_SSE2)
-  target_compile_options (${fftw3_lib} PRIVATE ${SSE2_FLAG})
-endif ()
-if (HAVE_AVX)
-  target_compile_options (${fftw3_lib} PRIVATE ${AVX_FLAG})
-endif ()
-if (HAVE_AVX2)
-  target_compile_options (${fftw3_lib} PRIVATE ${AVX2_FLAG})
-endif ()
-if (HAVE_FMA)
-  target_compile_options (${fftw3_lib} PRIVATE ${FMA_FLAG})
-endif ()
-if (HAVE_LIBM)
-  target_link_libraries (${fftw3_lib} m)
-endif ()
+  list (APPEND subtargets ${fftw3_lib})
 
-set (subtargets ${fftw3_lib})
+  if (ENABLE_THREADS AND Threads_FOUND)
+    if (WITH_COMBINED_THREADS)
+      target_link_libraries (${fftw3_lib} ${CMAKE_THREAD_LIBS_INIT})
+    else ()
+      add_library (${fftw3_lib}_threads ${fftw_threads_SOURCE})
+      target_link_libraries (${fftw3_lib}_threads ${fftw3_lib})
+      target_link_libraries (${fftw3_lib}_threads ${CMAKE_THREAD_LIBS_INIT})
+      list (APPEND subtargets ${fftw3_lib}_threads)
+    endif ()
+  endif ()
 
-if (Threads_FOUND)
-  if (WITH_COMBINED_THREADS)
-    target_link_libraries (${fftw3_lib} ${CMAKE_THREAD_LIBS_INIT})
-  else ()
-    add_library (${fftw3_lib}_threads ${fftw_threads_SOURCE})
-    target_link_libraries (${fftw3_lib}_threads ${fftw3_lib})
-    target_link_libraries (${fftw3_lib}_threads ${CMAKE_THREAD_LIBS_INIT})
-    list (APPEND subtargets ${fftw3_lib}_threads)
+  if (ENABLE_OPENMP AND OPENMP_FOUND)
+    add_library (${fftw3_lib}_omp ${fftw_omp_SOURCE})
+    target_link_libraries (${fftw3_lib}_omp ${fftw3_lib})
+    target_link_libraries (${fftw3_lib}_omp ${CMAKE_THREAD_LIBS_INIT})
+    list (APPEND subtargets ${fftw3_lib}_omp)
+    target_compile_options (${fftw3_lib}_omp PRIVATE ${OpenMP_C_FLAGS})
   endif ()
-endif ()
 
-if (OPENMP_FOUND)
-  add_library (${fftw3_lib}_omp ${fftw_omp_SOURCE})
-  target_link_libraries (${fftw3_lib}_omp ${fftw3_lib})
-  target_link_libraries (${fftw3_lib}_omp ${CMAKE_THREAD_LIBS_INIT})
-  list (APPEND subtargets ${fftw3_lib}_omp)
-  target_compile_options (${fftw3_lib}_omp PRIVATE ${OpenMP_C_FLAGS})
+  install(TARGETS ${fftw3_lib}
+            EXPORT FFTW3LibraryDepends
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+
+  # pkgconfig file
+  set (prefix ${CMAKE_INSTALL_PREFIX})
```

Comment:
I think this code, that doesn't depend on target, should be removed from function.

Reply:
Hm… It doesn't depend on the target, that's right, but those are the input variables for the following `configure_file(…)` call, so I think we should still keep them immediately before that call for readability. No damage done except for a couple CPU cycles.

Reply:
> those are the input variables

There is also code below:
```cmake
export (TARGETS ${fftw3_lib} NAMESPACE FFTW3:: FILE ${PROJECT_BINARY_DIR}/FFTW3LibraryDepends.cmake)
```

```cmake
install(EXPORT FFTW3LibraryDepends
        NAMESPACE FFTW3::
        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/fftw3${PREC_SUFFIX}
        COMPONENT Development)
```

Need to test if it will work correctly if called several times.

Reply:
That code below you quoted differs on each call of the function though. `${fftw3_lib}` evaluates to `fftw3`, `fftw3f` ,`fftw3l` or `fftw3q` depending on the precision selected.